app/controllers/have_read_books_controller.rb

```diff
@@ -2,15 +2,35 @@ class HaveReadBooksController < ApplicationController
   before_action :authenticate_user!
 
   def index
+    @current_month = Time.zone.now.strftime("%b")
+    @current_day = Time.zone.now.day
+    @current_year = Time.zone.now.year
+
     if sort_params.present?
-      @have_read_books = current_user.completed_books.order_by(sort_params).includes(:authors, :genres)
+      @have_read_books = current_user.completed_books.order_by(sort_params).includes(:authors, :genres)&.page params[:page]
+    else
+      @have_read_books = current_user.completed_books.order("have_read_books.date_completed desc").includes(:authors, :genres)&.page params[:page]
```

Comment:
This logic could probably be extracted into the order_by method actually since the only difference is the content in `order`.

Reply:
This would mean making the order_by method in the Book class specific to have_read_books. Or do you mean I should add several elsif statements and move the default logic (no sorting params) for books, have_read_books, and to_read_books into the order_by method?